src/main/api/catalog.ts

```diff
@@ -30,20 +31,28 @@ export const CATALOG_CONFIG_ID = "catalog";
 
 const NB_PUB = 10;
 
+const viewToR2Pub = memoizeWith((v) => v.identifier, (view: PublicationView) => {
```

Comment:
I like the idea of using Ramda's `memoizeWith()`, however this creates an in-memory cache of the entire `R2Publication` object for nearly all the publications in the catalog. Instead, couldn't we just memoize a `isPDF` boolean, as this is all we need `viewToR2Pub()` / `R2Publication` for.

Reply:
i'm wonder if we can use this for the whole app ?

Reply:
Well, let's see ... the memoization is really just a lazy-loaded memory map between string identifier and R2Publication instances, but over hundreds/thousands of publications in the catalog this will eventually create significant additional memory allocations / footprint.

Reply:
Addressed via https://github.com/edrlab/thorium-reader/pull/1292/commits/697b7977c22a98da66fb3168907404953546db51